Streams

kartelpol opened the pull request:
https://epa.ms/nodejs18-hw3

Commit:
add file watcher, importer module, csv to json converter

Commit:
apply review comments
	add jsdoc
	fix import module due to task (takes and import only one file)
	prettify logging and naming
	add logs to the gitignore

Commit:
add streams util
add predefined actions and CLI args validation
add to data/css stylesheets to test cssBundler action
add errors handling

Commit:
put error messages into config

deser commented on src/utils/streams/errorMsgs.js:
```diff
@@ -0,0 +1,8 @@
+        wrong_format: 'INVALID ARGUMENTS COMBINATION',
+        wrong_number: (arg) => `NEED AN EXTRA ARGUMENT --${arg}. SEE --HELP (-H)`,
+        wrong_extention: (ext) => `FORMAT ERROR. EXPECTED ${ext.toUpperCase} FILE`,
+        wrong_path: 'PATH IS NOT EXIST'
```
does not

deser commented on app.js:
```diff
@@ -1,9 +1,43 @@
+import Importer from './src/importer';
 
 logger.info(config.name);
+console.log(process.argv);
```
logger or console.log? :)

deser commented on config/webpack.config.js:
```diff
@@ -14,7 +14,8 @@ const clean = new CleanWebpackPlugin(['build'], {
 });
 
-module.exports = (env, argv) => ({
+module.exports = (env, argv) => {console.log(argv); return({
```
logger

deser commented on src/utils/streams/actions.js:
```diff
@@ -0,0 +1,131 @@
+const { Writable } = require('stream');
```
const { Writable,Transform } = require('stream');

deser commented on src/utils/streams/actions.js:
```diff
@@ -0,0 +1,131 @@
+const BUNDLE_CSS_NAME = 'bundle.css';
+const FINAL_CSS_NAME = 'end-bundle-with-this.css';
+
+function reverse() {
```
![image](https://user-images.githubusercontent.com/3010698/51492522-3fef8200-1dc3-11e9-87c3-d24c0639ee11.png)

Please, follow interface (at least, reverse should take string as argument)

kartelpol replied:
I just didn't understand why I need to read stdin outside the action functions and then pass inside chunks?

deser commented on src/utils/streams/actions.js:
```diff
@@ -0,0 +1,131 @@
+}
+
+function convertToFile(filePath) {
+    filePath = join(__dirname, filePath);
```
think how to eliminate code duplication between convertFromFile\convertToFile

deser commented on src/utils/streams/actions.js:
```diff
@@ -0,0 +1,131 @@
+    const outputFile = join(__dirname, relativeDirPath, BUNDLE_CSS_NAME);
+    dirPath = join(__dirname, relativeDirPath);
+
+    const styleStreams = await getCssStreamsFromDir(dirPath);
```
try\catch

deser commented on src/utils/streams/actions.js:
```diff
@@ -0,0 +1,131 @@
+}
+
+function getCssStreamsFromDir(dirPath) {
+    const styleStreams = [];
```
move it's definition inside promise

deser commented on src/utils/streams/actions.js:
```diff
@@ -0,0 +1,131 @@
+function getFinalCssStream() {
+    const requestedCSSPath = join(__dirname, FINAL_CSS_NAME);
+    try {
+        return readStream = createReadStream(requestedCSSPath);
```
```suggestion
        return createReadStream(requestedCSSPath);
```

deser commented on src/utils/streams/index.js:
```diff
@@ -0,0 +1,54 @@
+    if (process.argv[2] === '--help' || process.argv[2] === '-h' || process.argv.length == 2) {
+        parser.printHelp();
+    } else if (args.action && (process.argv[2].indexOf('--action=') == 0 || process.argv[2].indexOf('-a=') == 0)) {
+        checkAction(args);
```
pass only action name, not the whole `args`

kartelpol replied:
Would be better to use global args inside the checkAction?

deser reviewed (changes requested):
Good job. Please fix comments and write tests

deser commented on src/utils/streams/actions.js:
```diff
@@ -0,0 +1,131 @@
+}
+
+function transform() {
+    const upperCaseTransformingStream = new Transform({
```
they say you need to use `through2`

kartelpol replied:
It's just an example as I understood

Commit:
perform refactoring
	add streams bundling
	remove code duplication
	add through2 module to follow the interface
